src/exceptiongroup/_exceptions.py

```diff
@@ -121,9 +145,27 @@ def subgroup(self: T, __condition: _SplitCondition[EBase]) -> T | None:
         else:
             return None
 
+    @overload
+    def split(
+        self: Self,
+        __condition: type[_BaseExceptionT] | tuple[type[_BaseExceptionT], ...],
+    ) -> tuple[BaseExceptionGroup[_BaseExceptionT] | None, Self | None]:
+        ...
+
+    @overload
     def split(
-        self: T, __condition: _SplitCondition[EBase]
-    ) -> tuple[T | None, T | None]:
+        self: Self, __condition: Callable[[_BaseExceptionT_co], bool]
+    ) -> tuple[Self | None, Self | None]:
+        ...
+
+    def split(
+        self: Self,
+        __condition: type[_BaseExceptionT]
+        | tuple[type[_BaseExceptionT], ...]
+        | Callable[[_BaseExceptionT_co], bool],
+    ) -> tuple[BaseExceptionGroup[_BaseExceptionT] | None, Self | None] | tuple[
+        Self | None, Self | None
```

Comment:
I'm not seeing how we get a non-`Self` case from this, since we're using `.derive()`?

Reply:
I'm not sure; maybe my implementation misses a case? This is how it was annotated in the typeshed though, so :shrug:

Reply:
Hmm, might be a typeshed bug then?  I'd be fine with merging it as-is though; it doesn't seem particularly important to investigate.

Reply:
Ah, got it: the correct return-type annotation is `tuple[BaseExceptionGroup[_BaseExceptionT] | None, BaseExceptionGroup[_BaseExceptionT] | None]`, no use of `Self` or overrides.  That's because the more precise annotation is only true for subclasses which implement `.derive()`, in which case they should also ship more precise (but still not `Self`) annotations for `.subgroup()` and `.split()`.

I'll open an upstream issue against typeshed shortly.